[core][docs] Lint some top level core docs (#49703)
ray-project/ray: doc/source/ray-core/key-concepts.rst

Actors
------

Actors extend the Ray API from functions (tasks) to classes. An actor is essentially a stateful worker (or a service). When you instantiate a new actor, Ray creates a new worker and schedules methods of the actor on that specific worker. The methods can access and mutate the state of that worker. Like tasks, actors support CPU, GPU, and custom resource requirements.

See the :ref:`User Guide for Actors <actor-guide>`.

Objects
-------

> Tasks and actors create objects and compute on objects. You can refer to these objects as *remote objects* because they're stored anywhere in a Ray cluster, and you use *object refs* to refer to them. Ray caches remote objects in its distributed `shared-memory <https://en.wikipedia.org/wiki/Shared_memory>`__ *object store*, and there is one object store per node in the cluster. In the cluster setting, a remote object can live on one or many nodes, independent of who holds the object ref.

Reviewer: ```suggestion
Tasks and actors create objects and compute on objects. You can refer to these objects as *remote objects* because Ray stores them anywhere in a Ray cluster, and you use *object refs* to refer to them. Ray caches remote objects in its distributed `shared-memory <https://en.wikipedia.org/wiki/Shared_memory>`__ *object store*, and Ray creates one object store per node in the cluster. In the cluster setting, a remote object can live on one or many nodes, independent of who holds the object ref.
```

See the :ref:`User Guide for Objects <objects-in-ray>`.

Placement Groups
----------------

> Placement groups allow users to atomically reserve groups of resources across multiple nodes. They can be then used to schedule Ray tasks and actors packed as close as possible for locality (PACK), or spread apart (SPREAD). Placement groups are generally used for gang-scheduling actors, but also support tasks.

Reviewer: ```suggestion
Placement groups allow users to atomically reserve groups of resources across multiple nodes. Ray can then use them to schedule Ray tasks and actors packed as close as possible for locality (PACK), or spread apart (SPREAD). Use placement groups for gang-scheduling actors, but also support tasks.
```

See the :ref:`User Guide for Placement Groups <ray-placement-group-doc-ref>`.

Environment Dependencies
------------------------

> When Ray executes tasks and actors on remote machines, their environment dependencies (such as Python packages, local files, environment variables) must be available for the code to run. To address this problem, you can

Reviewer: ```suggestion
When Ray executes tasks and actors on remote machines, their environment dependencies, such as Python packages, local files, environment variables, must be available for Ray to run the code. To address this problem, you can
```
